extract_data.py

```diff
@@ -97,31 +96,85 @@ def extract_and_collect_data_from_generated_files(directory: str, has_dedicated_
             "frequency-hertz": None,
             "human_readable_frequency": None
         }
-
-    result = [chassis, mobo, cpu]
-    if wifi_cards:
-        result += wifi_cards
-
-    if isinstance(dimms, dict):
-        # otherwise it will append every key-value pair of the dict
-        result.append(dimms)
-    else:
+    """
+    def normalize_brands(coll_dict):
+        names = {}
+        with open("normalized.csv", "r") as f:
+            while True:
+                line = f.readline()
+                if line == "":
+                    break
+                k, v = line.split(";", 2)[0:2]
+                names[k] = v
+        for p_dict in coll_dict:
+            if "brand" in p_dict.keys():
+                if p_dict["brand"] in names.keys():
+                    p_dict["brand"] = names[p_dict["brand"]]
+
+    item_keys = ["arrival-batch", "cib", "cib-old", "cib-qr", "data-erased", "mac", "notes",
+                      "os-license-code", "os-license-version", "other-code", "owner", "smart-data",
+                      "sn", "software", "surface-scan", "working", "wwn"]
+    bmv = ["brand", "model", "variant"]
+
+    normalize_brands([chassis, mobo, cpu, gpu, psu]+dimms+wifi_cards+disks)
+
+    products = [chassis, mobo]
+# setting mobo dict
+    new_mobo = {"features": {k: v for k, v in mobo.items() if k in bmv+item_keys}, "contents": []}
+# mount the cpu
+    if cpu.__len__() != 0:
+        products.append(cpu)
+        new_mobo["contents"].append({"features": {k: v for k, v in cpu.items() if k in bmv+item_keys}})
+# adding some ram
+    if isinstance(dimms, list):
+        products += dimms
         for dimm in dimms:
-            result.append(dimm)
+            new_mobo["contents"].append({"features": {k: v for k, v in dimm.items() if k in bmv+item_keys}})
+    elif dimms.__len__() != 0:
```

Comment:
```suggestion
    elif len(dimms) != 0:
```
Also always use len(item) instead of item.__len__(), they are the same method but len() is more readable